spec/compiler/type_inference/global_spec.cr

```diff
@@ -185,6 +185,20 @@ describe "Global inference" do
       )) { types["Foo"] }
   end
 
+  it "doesn't infer from new if generic" do
+    assert_error %(
+      class Foo(T)
+        def self.new
+          10
+        end
+      end
+
+      $x = Foo.new
+      $x
+      ),
+      "undefined global variable '$x'"
```

Comment:
I don't quite get the problem here, if you override new you basically give up being able to create a new instance (unless you call allocate from within and return that). But why not allow it? In any case the error message is very misleading.